mne/preprocessing/maxwell.py

```diff
@@ -554,7 +490,7 @@ def _update_info(raw, origin, int_order, ext_order, nsens, nmoments):
     max_info_dict = dict(max_st={}, sss_cal={}, sss_ctc={},
                          sss_info=sss_info_dict)
 
-    block_id = dict(machid=-1 * np.ones(2), secs=-1, usecs=-1, version=-1)
+    block_id = dict(machid=get_machid(), secs=-1, usecs=-1, version=-1)
     proc_block = dict(max_info=max_info_dict, block_id=block_id,
                       creator='mne-python v%s' % __version__,
                       date=-1, experimentor='')
```

Comment:
I wouldn't expect all fields to actually *match*, especially things like `version` since we have a different piece of software altogether. Just take a look at all the fields that are populated on the `testing` dataset version, and populate them based on what makes sense for your new algorithm. I don't know for example what is in the `experimenter` field, but if it comes from the `raw.info` somewhere you can take it, but if `MaxFilter` sets it to e.g. `maxfilter 2.2` then you should do `'mne-python %s' % __version__` or so.

Regarding `secs` and `usecs`, see how we set those fields elsewhere in the package, if at all. Also check the manual. It might just be you can get it from `time.time()` with some simple arithmetic.